vicodyn/include/cocaine/vicodyn/balancer/utils.hpp

```diff
@@ -0,0 +1,25 @@
+#include <cstdlib>
+#include <vector>
+
+namespace cocaine {
+namespace vicodyn {
+namespace balancer {
+
+template<class Iterator, class Predicate>
+auto choose_random_if(Iterator begin, Iterator end, std::size_t size, Predicate p) -> Iterator {
+    std::vector<Iterator> chosen;
+    chosen.reserve(size);
+    for(; begin != end; ++begin) {
+        if(p(*begin)) {
+            chosen.push_back(begin);
+        }
+    }
+    if(chosen.empty()) {
+        return end;
+    }
+    return chosen[rand() % chosen.size()];
```

Comment:
Is it rand from `std` namespace? Consider using `std::` namespace prefix in code, where appropriate.

`std::rand` returns value in interval [0, RAND_MAX], is it any warranties about RAND_MAX? Seems practically impossible, but could it be some selection alignment in case when (magically) RAND_MAX < chosen.size()?

Reply:
> Consider using std:: namespace prefix in code, where appropriate.

Agree, it's copypaste. I'll fix it.

cppreference.com:
> It's guaranteed that this value is at least 32767

RAND_MAX equals INT_MAX in our production environment.
Do you think it's necessary to implement hard logic for supporting very large stl-container sizes?

Reply:
@karitra, look at aca75eb